src/hotspot/share/utilities/exceptions.cpp

```diff
@@ -82,10 +82,17 @@ void ThreadShadow::clear_pending_nonasync_exception() {
 
 // Implementation of Exceptions
 
-bool Exceptions::special_exception(JavaThread* thread, const char* file, int line, Handle h_exception) {
+bool Exceptions::special_exception(JavaThread* thread, const char* file, int line, Handle h_exception, Symbol* h_name, const char* message) {
```

Comment:
So IIUC now we either have a non-null exception and a null symbol and msg; or vice-versa. Can we assert that so someone doesn't mistakenly pass in non-null values for all three.

Reply:
Sure: https://github.com/openjdk/jdk/pull/16401/commits/7e6c1a63f4c5104cb82b4bad38c9b8910b156127